### PR TITLE
Feature/setup cypress for e2e testing

### LINKED ISSUE #154: Setup Cypress for E2E testing
**Is your enhancement related to a problem? Please describe.**

See overall concept in the [OSBP guide](https://10up.github.io/Open-Source-Best-Practices/testing/#e2e-testing) and example setups on [Simple Podcasting](https://github.com/10up/simple-podcasting), [Restricted Site Access](https://github.com/10up/restricted-site-access), and https://github.com/10up/ElasticPress/pull/2446.

**Describe the solution you'd like**

- [ ]   add base GitHub Action
- [ ]   add .wp-env.json
- [ ]   add base docker files
- [ ]   update package.json / package lock files
- [ ]   add cypress tests
- [ ]   remove any WP Acceptance components

**Designs**

n/a

**Describe alternatives you've considered**

n/a

**Additional context**

https://github.com/10up/cypress-wp-setup/ && https://github.com/10up/cypress-wp-utils will be helpful in getting this spun up and some initial test commands to leverage.

### PR DESCRIPTION
**Pull request by barneyjeffries**
### Description of the Change
This PR adds Cypress for e2e testing, configured and with 2 tests set-up:

1. settings.test - the test adds the admin credentials for a test azure account and uploads a file to verify integration is working
2. admin.test - the test logs in and deactivates and activates the plugin

NOTE: We need to add the test azure account credentials to the GitHub repo but I don't have permissions to do this. The constants should be set as:
MICROSOFT_AZURE_ACCOUNT_NAME
MICROSOFT_AZURE_ACCOUNT_KEY
MICROSOFT_AZURE_CONTAINER

Closes #154

### How to test the Change

1. Pull changes
2. Run npm install
4. Build assets: npm run build
5. Start local environment: npm run env:start
6. Open Cypress for testing supplying test credentials:
`CYPRESS_MICROSOFT_AZURE_ACCOUNT_NAME=[accountname] CYPRESS_MICROSOFT_AZURE_ACCOUNT_KEY=[key] CYPRESS_MICROSOFT_AZURE_CONTAINER=[container] npm run cypress:open`
7. Use Cypress GUI to run tests and verify they are working

### Changelog Entry

> Added - New feature

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

**Comment by iamdharmesh** [comment]
@barneyjeffries Thanks a lot for the PR.

@jeffpaul could you please help with adding Azure credentials to GH action secrets to the repo? I tried running tests locally and it seems working fine.

Thanks.

**Comment by jeffpaul** [comment]
@iamdharmesh I've added what I believe are the necessary secrets, but please let me know if anything isn't acting as expected and I'll research further... thanks!

**Comment by jeffpaul** [comment]
@iamdharmesh bumping this back up to see if the repo secrets allow this to function as you'd expect?

File: .github/workflows/cypress.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    env:
+      MICROSOFT_AZURE_ACCOUNT_NAME: ${{ secrets.MICROSOFT_AZURE_ACCOUNT_NAME }}
+      MICROSOFT_AZURE_ACCOUNT_KEY: ${{ secrets.MICROSOFT_AZURE_ACCOUNT_KEY }}
+      MICROSOFT_AZURE_CONTAINER: ${{ secrets.MICROSOFT_AZURE_CONTAINER }}
+      MICROSOFT_AZURE_USE_FOR_DEFAULT_UPLOAD: ${{ secrets.MICROSOFT_AZURE_USE_FOR_DEFAULT_UPLOAD }}
+    strategy:
+      matrix:
+        core:
+          - {name: 'WP latest', version: 'latest'}
+          - {name: 'WP minimum', version: 'WordPress/WordPress#5.2'}
```
Comment (iamdharmesh) [inline .github/workflows/cypress.yml]:
```suggestion
          - {name: 'WP minimum', version: 'WordPress/WordPress#5.7'}
```

**Comment by iamdharmesh** [comment]
@jeffpaul Thanks for adding repo secrets. it's working fine after some minor yml file changes.

**Review by iamdharmesh — APPROVED** [review]
Thanks @barneyjeffries, LGTM 🚀

**Pull request merged**